File: src-tauri/src/ani/ipc.rs
use crate::ani::connect;

#[tauri::command]
pub async fn login(username: &str, password: &str) -> Result<String, String> {
    let udp_client = connect::get_udp_client();
    udp_client
        .login(username, password)
        .map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn query(name: &str) -> Result<String, String> {
    let udp_client = connect::get_udp_client();
    udp_client.query(name).map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn logout() -> Result<String, String> {
    let udp_client = connect::get_udp_client();
    udp_client.logout().map_err(|e| e.to_string())
}

#[tauri::command]
pub async fn updated() -> Result<String, String> {
    let udp_client = connect::get_udp_client();
    udp_client.updated().map_err(|e| e.to_string())
}

File: src/ipc.ts
import { invoke } from '@tauri-apps/api';

export interface LoginResult {
  raw: string;
}

export async function login(username: string, password: string): Promise<string> {
  return invoke('login', { username, password }) as Promise<string>;
}

export async function query(name: string): Promise<string> {
  return invoke('query', { name }) as Promise<string>;
}

export async function logout(): Promise<string> {
  return invoke('logout', {}) as Promise<string>;
}

export async function updated(): Promise<string> {
  return invoke('updated', {}) as Promise<string>;
}
